cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/CgmesExport.java

```diff
@@ -27,22 +38,50 @@
 
     @Override
     public void export(Network network, Properties params, DataSource ds) {
-
-        // Right now the network must contain the original CgmesModel
-        // In the future it should be possible to export to CGMES
-        // directly from an IIDM Network,
-        // without the need for the original CgmesModel
+        Objects.requireNonNull(network);
         CgmesModelExtension ext = network.getExtension(CgmesModelExtension.class);
-        if (ext == null) {
-            throw new CgmesModelException("No extension for CGMES model found in Network");
+        if (params != null && Boolean.valueOf(params.getProperty("cgmes.export.usingOnlyNetwork"))) {
+            if (ext != null) {
+                CgmesModel cgmesSource = ext.getCgmesModel();
+                if (cgmesSource != null) {
+                    throw new CgmesModelException("CGMES model should not be available as Network extension");
```

Comment:
Not sure why an exception is thrown? We could just ignore it.

Reply:
Check has been removed.